src/docs/sphinx/dev_guide/gpu_porting.rst

```diff
@@ -0,0 +1,318 @@
+.. ## other Axom Project Developers. See the top-level LICENSE file for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+
+.. _gpu-porting-label:
+
+*******************************
+GPU Porting in Axom
+*******************************
+
+Axom uses RAJA and Umpire as the main workhorses for GPU porting,
+with a set of convenience macros, `axom`-namespaced wrappers around commonly
+used functions, and preset execution spaces for host/device execution.
+
+For the user's guide on using GPU utilities, see also
+:ref:`Core Acceleration<core-acceleration>`
+
+
+.. _gpu-macros-label:
+
+===============
+Macros 
+===============
+
+Axom's macros can be found in
+`axom/core/Macros.hpp <https://github.com/LLNL/axom/blob/develop/src/axom/core/Macros.hpp>`_
+
+Most of the GPU-related macros are used to guard device code before compilation
+time or for ``__host__ __device__`` decoration of functions/lambdas.
+
+For guarding device code:
+
+* ``AXOM_USE_GPU`` - Defined if Axom is built with CUDA or HIP
+
+	* ``AXOM_USE_CUDA`` - Defined if Axom is built with CUDA
+	* ``AXOM_USE_HIP`` - Defined if Axom is built with HIP
+
+* ``AXOM_DEVICE_CODE`` - Denotes code will be compiled on device
+* ``AXOM_GPUCC`` - Denotes code will be compiled with HIP or CUDA compiler
+
+For ``__host__ __device__`` decoration::
+
+  //---------------------------------------------------------------------------
+  // Functions
+  //---------------------------------------------------------------------------
+
+  #define AXOM_DEVICE __device__
+
+  #define AXOM_HOST_DEVICE __host__ __device__
+
+  #define AXOM_HOST __host__
+
+  //---------------------------------------------------------------------------
+  // Lambdas
+  //---------------------------------------------------------------------------
+
+  #define AXOM_LAMBDA [=] AXOM_HOST_DEVICE
+
+  #define AXOM_DEVICE_LAMBDA [=] AXOM_DEVICE
+
+  #define AXOM_HOST_LAMBDA [=] AXOM_HOST
+
+
+.. _gpu-memory-label:
+
+===============
+Memory 
+===============
+
+Axom's memory management functionality can be found in
+`axom/core/memory_management.hpp <https://github.com/LLNL/axom/blob/develop/src/axom/core/memory_management.hpp>`_
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Memory Management Routines
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Umpire has the concept of "allocators" associated with each
+`memory resource type`_ (``umpire::resource::MemoryResourceType``)
+
+To allocate memory on a particular resource, you use the ID for the allocator
+associated with the MemoryResourceType.
+
+You are able to set a default allocator, where all your memory allocations will
+go on the resource associated with the allocator unless otherwise specified::
+
+  //---------------------------------------------------------------------------
+  // Getters and Setters for Allocators and ID
+  //---------------------------------------------------------------------------
+
+  // Gets the allocator ID associated with the given MemoryResourceType
```

Comment:
It may be better and easier to maintain if code comments are used to document code blocks like this. That way the docs won't ned to change when the code does.